Utils/WeakList.swift

```diff
@@ -0,0 +1,66 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import Foundation
+
+/**
+ * A list that weakly holds references to its items.
+ * Note that while the references themselves are cleared, their wrapper objects
+ * are not (though they are reused). Also note that since slots are reused,
+ * order is not preserved.
+ *
+ * This class crashes at runtime with EXC_BAD_ACCESS if a protocol is given as
+ * the type T. Make sure to use a class type.
+ */
+public class WeakList<T: AnyObject>: SequenceType {
+    private var items = [WeakRef<T>]()
+
+    public init() {}
+
+    /**
+     * Adds an item to the list.
+     * Note that every insertion iterates through the list to find any "holes" (items that have
+     * been deallocated) to reuse them, so this class may not be appropriate in situations where
+     * appending is frequent.
+     */
+    public func add(item: T) {
```

Comment:
Hmm... Good point about the ordering, although I don't think I really thought of append that way (or insert for that matter). I think I'd still take some consistency over a new different api. I've been annoyed enough with count(), size() and length() in Java.